Sources/NetworkProtection/Networking/NetworkProtectionClient.swift

```diff
@@ -185,16 +205,38 @@ public final class NetworkProtectionBackendClient: NetworkProtectionClient {
         }
     }
 
-    public func redeem(inviteCode: String) async -> Result<String, NetworkProtectionClientError> {
-        let requestBody = RedeemRequestBody(code: inviteCode)
+    public func authenticate(withMethod method: NetworkProtectionAuthenticationMethod) async -> Result<String, NetworkProtectionClientError> {
+        switch method {
+        case .inviteCode(let code):
+            return await redeem(inviteCode: code)
+        case .subscription(let accessToken):
+            return await exchange(accessToken: accessToken)
+        }
+    }
+
+    private func redeem(inviteCode: String) async -> Result<String, NetworkProtectionClientError> {
```

Comment:
I wonder if this would be a bit simpler if we just exposed these functions instead of having the enum. I feel like it would be a bit less code. Not a blocker though.